submissions/AnnaStoyano/memory-pair-game/index.js

```diff
@@ -0,0 +1,152 @@
+class Game {
+    constructor(cards) {
+        this.container = document.querySelector('.container');
+        this.cards = cards;
+    }
+
+    render() {
+        this.cards.forEach(card => this.container.insertAdjacentElement('beforeend', card.render()));
+    }
+
+    checkEqualCards(activeCards) {
+        const activeCardsNames = activeCards.map(card => card.cardForDisplay.dataset.name);
+        if (activeCardsNames[0] === activeCardsNames[1]) {
+            activeCards.forEach(card => card.addOpened());
+            return true;
+        }
+        return false;
+    }
+
+    removeActiveCards(cards, isOpened) {
+        let time = isOpened ? 0 : 600;
+        setTimeout(function () {
+            cards.forEach(card => card.removeActive());
+        }, time);
+        return [];
+    }
+
+    processClickOnCards() {
+        const game = this;
+        this.container.addEventListener('click', function (e) {
+            let activeCards = game.getActiveCards();
+            let target = e.target;
+            let targetCard = target.parentElement;
+            if (activeCards.length < 2 && targetCard.classList.contains('card')) {
+                if (!targetCard.classList.contains("opened")) {
+                    game.findCard(targetCard).addActive();
+                    activeCards = game.getActiveCards();
+                }
+            }
+            if (activeCards.length === 2) {
+                let isOpened = game.checkEqualCards(activeCards);
+                activeCards = game.removeActiveCards(activeCards, isOpened);
+            }
+            game.checkWin.bind(game)();
+        }, false);
+    }
+
+    shuffleCard() {
+        this.cards.sort(() => 0.5 - Math.random());
+    }
+
+    startGame() {
+        this.shuffleCard();
+        this.render();
+        this.processClickOnCards();
+    }
+
+    restartGame() {
+        this.shuffleCard();
+        this.render();
+    }
+
+    findCard(cardHTMLelement) {
+        let card = this.cards.find(card => card.cardForDisplay.id == cardHTMLelement.id);
```

Comment:
You can simplify this functions by not creation of the `card` variable
```suggestion
        return this.cards.find(card => card.cardForDisplay.id == cardHTMLelement.id);
```